Main.elm

```diff
@@ -51,30 +55,50 @@ main =
   StartApp.Simple.start { model = queryParams, view = view, update = update }
 
 
+withBlankDefault : Maybe Param -> Param
+withBlankDefault =
+  Maybe.withDefault ( "", "" )
+
+
+safeGetAtIndex : Model -> Int -> Param
+safeGetAtIndex model index =
+  List.Extra.getAt model index
+    |> withBlankDefault
+
+
 update : Action -> Model -> Model
 update action model =
   case action of
     UpdateKey ( index, newKey ) ->
       let
-        current =
-          List.Extra.getAt model index
-            |> Maybe.withDefault ( "", "" )
-
-        new =
-          ( newKey, snd current )
+        ( _, value ) =
```

Comment:
In Elm we can destructure in assignments just like in ES2105, but also use `_` to denote that we don't care about that particular value.